cupy/cuda/nccl.pyx

```diff
@@ -29,48 +29,76 @@ cdef extern from "cupy_nccl.h":
     void ncclCommDestroy(ncclComm_t comm)
     ncclResult_t ncclCommCuDevice(const ncclComm_t comm, int* device)
     ncclResult_t ncclCommUserRank(const ncclComm_t comm, int* rank)
-    ncclResult_t ncclAllReduce(const void* sendbuff, void* recvbuff, int count,
+    ncclResult_t ncclAllReduce(const void* sendbuff, void* recvbuff,
+                               size_t count,
                                ncclDataType_t datatype, ncclRedOp_t op,
                                ncclComm_t comm, driver.Stream stream)
-    ncclResult_t  ncclReduce(const void* sendbuff, void* recvbuf, int count,
+    ncclResult_t  ncclReduce(const void* sendbuff, void* recvbuf, size_t count,
                              ncclDataType_t datatype, ncclRedOp_t op, int root,
                              ncclComm_t comm, driver.Stream stream)
-    ncclResult_t  ncclBcast(void* buff, int count, ncclDataType_t datatype,
+    ncclResult_t  ncclBcast(void* buff, size_t count, ncclDataType_t datatype,
                             int root, ncclComm_t comm, driver.Stream stream)
 
+    int NCCL_VERSION
+
+
+cdef dict ERROR1 = {
+    0: 'NCCL_ERROR_SUCCESS',
+    1: 'NCCL_ERROR_UNHANDLED_CUDA_ERROR',
+    2: 'NCCL_ERROR_SYSTEM_ERROR',
+    3: 'NCCL_ERROR_INTERNAL_ERROR',
+    4: 'NCCL_ERROR_INVALID_DEVICE_POINTER',
+    5: 'NCCL_ERROR_INVALID_RANK',
+    6: 'NCCL_ERROR_UNSUPPORTED_DEVICE_COUNT',
+    7: 'NCCL_ERROR_DEVICE_NOT_FOUND',
+    8: 'NCCL_ERROR_INVALID_DEVICE_INDEX',
+    9: 'NCCL_ERROR_LIB_WRAPPER_NOT_SET',
+    10: 'NCCL_ERROR_CUDA_MALLOC_FAILED',
+    11: 'NCCL_ERROR_RANK_MISMATCH',
+    12: 'NCCL_ERROR_INVALID_ARGUMENT',
+    13: 'NCCL_ERROR_INVALID_TYPE',
+    14: 'NCCL_ERROR_INVALID_OPERATION',
+}
 
-cdef dict STATUS = {
-    0: 'NCCL_STATUS_SUCCESS',
-    1: 'NCCL_STATUS_UNHANDLED_CUDA_ERROR',
-    2: 'NCCL_STATUS_SYSTEM_ERROR',
-    3: 'NCCL_STATUS_INTERNAL_ERROR',
-    4: 'NCCL_STATUS_INVALID_DEVICE_POINTER',
-    5: 'NCCL_STATUS_INVALID_RANK',
-    6: 'NCCL_STATUS_UNSUPPORTED_DEVICE_COUNT',
-    7: 'NCCL_STATUS_DEVICE_NOT_FOUND',
-    8: 'NCCL_STATUS_INVALID_DEVICE_INDEX',
-    9: 'NCCL_STATUS_LIB_WRAPPER_NOT_SET',
-    10: 'NCCL_STATUS_CUDA_MALLOC_FAILED',
-    11: 'NCCL_STATUS_RANK_MISMATCH',
-    12: 'NCCL_STATUS_INVALID_ARGUMENT',
-    13: 'NCCL_STATUS_INVALID_TYPE',
-    14: 'NCCL_STATUS_INVALID_OPERATION',
+cdef dict ERROR2 = {
+    0: 'NCCL_ERROR_SUCCESS',
+    1: 'NCCL_ERROR_UNHANDLED_CUDA_ERROR',
+    2: 'NCCL_ERROR_SYSTEM_ERROR',
+    3: 'NCCL_ERROR_INTERNAL_ERROR',
+    4: 'NCCL_ERROR_INVALID_ARGUMENT',
+    5: 'NCCL_ERROR_INVALID_USAGE',
 }
 
 
-class NcclError(RuntimeError):
+class Nccl1Error(RuntimeError):
```

Comment:
Yes, we can merge two classes `Nccl1Error` and `Nccl2Error` into `NcclError`. will update so.